python/lsst/cp/pipe/makeBrighterFatterKernel.py

```diff
@@ -0,0 +1,1380 @@
+        dataRef.dataId['visit'] = v2
+        exp2 = self.isr.runDataRef(dataRef).exposure
+        del dataRef.dataId['visit']
+        exps = [exp1, exp2]
+        self._checkExpLengthEqual(exp1, exp2, v1, v2)
+
+        detector = exps[0].getDetector()  # note we lose the detector in the next line as they belong to exp
+        ims = [self._convertImagelikeToFloatImage(exp) for exp in exps]
+
+        if self.debug.display:
+            self.disp1.mtv(ims[0], title=str(v1))
+            self.disp2.mtv(ims[1], title=str(v2))
+
+        sctrl = afwMath.StatisticsControl()
+        sctrl.setNumSigmaClip(sigma)
+        for imNum, im in enumerate(ims):
+
+            # calculate the sigma-clipped mean, excluding the borders
+            # safest to apply borders to all amps regardless of edges
+            # easier, camera-agnostic, and mitigates potentially dodgy
+            # overscan-biases around edges as well
+            for amp in detector:
+                ampName = amp.getName()
+                ampIm = im[amp.getBBox()]
+                mean = afwMath.makeStatistics(ampIm[border:-border, border:-border, afwImage.LOCAL],
+                                              afwMath.MEANCLIP, sctrl).getValue()
+                if ampName not in ampMeans.keys():
+                    ampMeans[ampName] = []
+                ampMeans[ampName].append(mean)
+                ampIm -= mean
+
+        diff = ims[0].clone()
+        diff -= ims[1]
+
+        temp = diff[border:-border, border:-border, afwImage.LOCAL]
+
+        # Subtract background. It should be a constant,
+        # but it isn't always (e.g. some SuprimeCam flats)
+        # TODO: Check how this looks, and if this is the "right" way to do this
+        binsize = self.config.backgroundBinSize
+        nx = temp.getWidth()//binsize
+        ny = temp.getHeight()//binsize
+        bctrl = afwMath.BackgroundControl(nx, ny, sctrl, afwMath.MEANCLIP)
+        bkgd = afwMath.makeBackground(temp, bctrl)
+
+        box = diff.getBBox()
+        box.grow(-border)
+        diff[box, afwImage.LOCAL] -= bkgd.getImageF(afwMath.Interpolate.CUBIC_SPLINE,
+                                                    afwMath.REDUCE_INTERP_ORDER)
+
+        variances = {}
+        coVars = {}
+        for amp in detector:
+            ampName = amp.getName()
+
+            diffAmpIm = diff[amp.getBBox()].clone()
+            diffAmpImCrop = diffAmpIm[border:-border-maxLag, border:-border-maxLag, afwImage.LOCAL]
+            diffAmpImCrop -= afwMath.makeStatistics(diffAmpImCrop, afwMath.MEANCLIP, sctrl).getValue()
+            w, h = diffAmpImCrop.getDimensions()
+            xcorr = np.zeros((maxLag + 1, maxLag + 1), dtype=np.float64)
+
+            # calculate the cross-correlation
+            for xlag in range(maxLag + 1):
+                for ylag in range(maxLag + 1):
+                    dim_xy = diffAmpIm[border+xlag:border+xlag + w, border+ylag: border+ylag + h, afwImage.LOCAL].clone()
```

Comment:
This line is too long and the formatting is confusing and not compliant with our standards (e.g. spaces around "+"). I suggest:
```
                    dim_xy = diffAmpIm[border + xlag: border + xlag + w,
                                       border + ylag: border + ylag + h,
                                       afwImage.LOCAL].clone()
```